modules/client/shared/src/main/scala/com/azavea/stac4s/api/client/StacClient.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.azavea.stac4s.api.client
+
+import com.azavea.stac4s._
+
+import eu.timepit.refined.types.string.NonEmptyString
+
+/** TODO: instead of returning F[List] we can return fs2.Stream */
+trait StacClient[F[_]] {
+  type Filter
+  def search(filter: Filter): F[List[StacItem]]
+  def collections: F[List[StacCollection]]
+  def collection(collectionId: NonEmptyString): F[Option[StacCollection]]
+  def items(collectionId: NonEmptyString): F[List[StacItem]]
+  def item(collectionId: NonEmptyString, itemId: NonEmptyString): F[Option[StacItem]]
+  def itemCreate(collectionId: NonEmptyString, item: StacItem): F[StacItem]
```

Comment:
@jisantuc @CloudNiner an API question, does it make sense to pass `collectionId` into the item creation function? Method would throw (Franklin rrror) right now if `item.collection` doesn't match `collectionId`. 

Mb to simplify this method signature? `def itemCreate(item: StacItem): F[StacItem]`?
Or to change the behavior to insert any time passed into the `collectionId` specified in the function declaration.

Reply:
IMO we'd leave the method signature as is so that the method params explicitly match the required params for the HTTP endpoint.

Reply:
@CloudNiner so it is up to user that `item.collection` doesn't correspond to the `collectionId` passed?

Reply:
Yes. It seems better to "fail safe" in that by forcing the match we prevent accidental submissions of items to incorrect collections than the other way around.

Reply:
One caveat though, if the error message for this case isn't currently clear, we should make sure it is so the user can easily fix.

Reply:
@CloudNiner nope, it is a totally clear exception message (smth like "item.collectionId doesnt match the target collectionId" but in a better language); Franklin handles it really great.